BUG: 1.10.0rc2 Windows wheel tests runs all segfault

The wheel build jobs are here: https://github.com/scipy/scipy/actions/runs/3753070381

The non-informative failures were originally discussed here, and Ralf and I kind of went for trying to build the wheels anyway, because it wasn't clear there was a genuine SciPy issue: https://github.com/scipy/scipy/pull/17640#issuecomment-1362081751

We have just slightly more information from the `cibuildwheel` testing now, but not too much more--we now know it is a segfault:

```
 D:\a\scipy\scipy/tools/wheels/cibw_test_command.sh: line 11:  1391 Segmentation fault      python -c "import sys; import scipy; sys.exit(not scipy.test())"
```

For now, I'm tentatively labeling this is an upstream bug since it started with NumPy `1.24.0`.

Two solutions I can think of:
- try to identify the problematic test and skip it with NumPy >= 1.24.0 pending further investigation, on Windows only
- (easier, I think?): pin the runtime testing for Windows wheel builds on GitHub actions to NumPy < 1.24.0 temporarily

Yeah, there is, I tried it on my fork here: https://github.com/tylerjereddy/scipy/pull/64

We get a bit more info, pasted below, perhaps a collection stage error/issue of some sort. There's a way to login to GitHub actions runners with a specific addon/action thingy, I've done that before.. could try that perhaps.

```
Run mkdir tmp

    Directory: D:\a\scipy\scipy

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----          12/22/[20](https://github.com/tylerjereddy/scipy/actions/runs/3754374273/jobs/6378537293#step:14:21)22  2:41 AM                tmp
============================= test session starts =============================
platform win32 -- Python 3.9.13, pytest-7.2.0, pluggy-1.0.0 -- C:\hostedtoolcache\windows\Python\3.9.13\x64\python.exe
cachedir: .pytest_cache
rootdir: D:\a\scipy\scipy, configfile: pytest.ini
plugins: timeout-2.1.0, xdist-3.1.0
collecting ... 
Error: Process completed with exit code 1.
```

The best I've been able to do is keyboard interrupt during the test collection phase where the segfault occurs (I can see the errors accumulating in the pytest counter so I press ctrl+c during the collection). I guess we'd need to believe that my local segfault/errors reflect the true situation on the runners vs. say me not working on Windows very often. If we believe that, then here's a sample error from the collection phase:

<details>

```
_______________________________________________________________ ERROR collecting stats/tests/test_contingency.py ________________________________________________________________
ImportError while importing test module 'C:\Users\treddy\python310_venv_windows\lib\site-packages\scipy\stats\tests\test_contingency.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\..\..\python310_venv_windows\lib\site-packages\_pytest\python.py:618: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
..\..\..\python310_venv_windows\lib\site-packages\_pytest\pathlib.py:533: in import_path
    importlib.import_module(module_name)
C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\lib\importlib\__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:992: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:992: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:883: in exec_module
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
..\..\..\python310_venv_windows\lib\site-packages\scipy\stats\__init__.py:485: in <module>
    from ._stats_py import *
..\..\..\python310_venv_windows\lib\site-packages\scipy\stats\_stats_py.py:53: in <module>
    from ._hypotests import _all_partitions
..\..\..\python310_venv_windows\lib\site-packages\scipy\stats\_hypotests.py:13: in <module>
    from scipy.fft import ifft
..\..\..\python310_venv_windows\lib\site-packages\scipy\fft\__init__.py:92: in <module>
    from ._helper import next_fast_len
E   ImportError: cannot import name 'next_fast_len' from 'scipy.fft._helper' (C:\Users\treddy\python310_venv_windows\lib\site-packages\scipy\fft\_helper.py)
```
</details>

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -46,7 +46,7 @@ jobs:
           gfortran --version
       - name: pip-packages
         run: |
-          pip install numpy==1.22.2 cython pybind11 pythran meson ninja pytest pytest-xdist pytest-timeout pooch
+          pip install numpy==1.22.2 cython "pybind11!=2.10.2" pythran meson ninja pytest pytest-xdist pytest-timeout pooch
       - name: openblas-libs
         run: |
           # Download and install pre-built OpenBLAS library

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -12,7 +12,7 @@ build-backend = 'mesonpy'
 requires = [
     "meson-python>=0.9.0",
     "Cython>=0.29.32",
-    "pybind11>=2.10.0",
+    "pybind11>=2.10.0,!=2.10.2",
     "pythran>=0.12.0",
     # `wheel` is needed for non-isolated builds, given that `meson-python`
     # doesn't list it as a runtime requirement (at least in 0.5.0)

